docs/public/deploying-a-devfile-using-odo.adoc

```diff
@@ -0,0 +1,332 @@
+  ✓  Checking files for pushing [1ms]
+  ✓  Syncing files to the component [839ms]
+  ✓  Executing devbuild command "npm install" [3s]
+  ✓  Executing devrun command "nodemon app.js" [2s]
+  ✓  Push devfile component mynodejs [33s]
+  ✓  Changes successfully pushed to component
+----
+
+. List the URLs of the component:
++
+[source,sh]
+----
+ $ odo url list
+     Found the following URLs for component mynodejs
+     NAME              URL                                       PORT     SECURE
+     mynodejs-8080     http://mynodejs-8080.apps-crc.testing     8080     false
+----
+
+. View your deployed application using the generated URL:
++
+[source,sh]
+----
+   $ curl http://mynodejs-8080.apps-crc.testing
+----
+
+== Deploying a Java Spring Boot® component locally to Docker
+
+In this example, we will be deploying the same Java Spring Boot® component we did earlier, but to a locally running Docker instance.
+
+*Prerequisites:* Docker `17.05` or higher installed
+
+. Enabling the separate pushtarget preference:
++
+[source,sh]
+----
+ $ odo preference set pushtarget docker
+ Global preference was successfully updated
+----
++
+You can configure a separate push target by making use of the `pushtarget` preference.
+
+. Create a component configuration of Spring Boot component-type named mydockerspringboot:
++
+[source,sh]
+----
+  $ odo create java-spring-boot mydockerspringboot
+  Experimental mode is enabled, use at your own risk
+
+  Validation
+   ✓  Checking devfile compatibility [26759ns]
+   ✓  Validating devfile component [75889ns]
+
+  Please use odo push command to create the component with source deployed
+----
+
+. Create a URL in order to access the deployed component:
++
+[source,sh]
+----
+ $ odo url create --port 8080
+  ✓  URL local-mydockerspringboot-8080 created for component: mydockerspringboot with exposed port: 37833
+
+ To apply the URL configuration changes, please use odo push
+----
++
+In order to access the docker application, exposed ports are required and automatically generated by odo.
+
+. Deploy the Spring Boot devfile component to Docker:
++
+[source,sh]
+----
+  $ odo push
+   •  Push devfile component mydockerspringboot  ...
+   ✓  Pulling image maysunfaisal/springbootbuild [601ms]
+
+  Applying URL configuration
+   ✓  URL 127.0.0.1:37833 created
+   ✓  Starting container for maysunfaisal/springbootbuild [550ms]
+   ✓  Pulling image maysunfaisal/springbootruntime [581ms]
+
+  Applying URL configuration
+   ✓  URL 127.0.0.1:37833 created
+   ✓  Starting container for maysunfaisal/springbootruntime [505ms]
+   ✓  Push devfile component mydockerspringboot [2s]
+   ✓  Changes successfully pushed to component
+----
++
+When odo deploys a devfile component, it pulls the images for each `dockercontainer` in `devfile.yaml` and deploys them.
++
+Each docker container that is deployed is labeled with the name of the odo component, linking all of them together.
```

Comment:
assuming here the `linking all of them together` is metaphorical linking and not literal `odo link`? this might confuse some people?